examples/google-esp-integration/with_sdk_v3/public/ads.js

```diff
@@ -0,0 +1,217 @@
+// Copyright 2013 Google Inc. All Rights Reserved.
+// You may study, modify, and use this example for any purpose.
+// Note that this example is provided "as is", WITHOUT WARRANTY
+// of any kind either expressed or implied.
+
+let adsManager;
+let adsLoader;
+let adDisplayContainer;
+let intervalTimer;
+let isAdPlaying;
+let isContentFinished;
+let playButton;
+let videoContent;
+
+/**
+ * Initializes IMA setup.
+ */
+function init() {
+  videoContent = document.getElementById("contentElement");
+  playButton = document.getElementById("playButton");
+  playButton.addEventListener("click", playAds);
+  setUpIMA();
+}
+
+/**
+ * Sets up IMA ad display container, ads loader, and makes an ad request.
+ */
+function setUpIMA() {
+  // Create the ad display container.
+  createAdDisplayContainer();
+
+  // Create ads loader.
+  adsLoader = new google.ima.AdsLoader(adDisplayContainer);
+  // Listen and respond to ads loaded and error events.
+  adsLoader.addEventListener(
+    google.ima.AdsManagerLoadedEvent.Type.ADS_MANAGER_LOADED,
+    onAdsManagerLoaded,
+    false
+  );
+  adsLoader.addEventListener(
+    google.ima.AdErrorEvent.Type.AD_ERROR,
+    onAdError,
+    false
+  );
+
+  // An event listener to tell the SDK that our content video
+  // is completed so the SDK can play any post-roll ads.
+  const contentEndedListener = function () {
+    // An ad might have been playing in the content element, in which case the
+    // content has not actually ended.
+    if (isAdPlaying) return;
+    isContentFinished = true;
+    adsLoader.contentComplete();
+  };
+  videoContent.onended = contentEndedListener;
+
+  // Request video ads.
+  const adsRequest = new google.ima.AdsRequest();
+  adsRequest.adTagUrl = "${AD_TAG_URL}";
```

Comment:
for future reference, this is the only non-formatting difference compared to google's code: https://github.com/googleads/googleads-ima-html5/blob/main/simple/ads.js